swap_meet/vendor.py

```diff
@@ -1,2 +1,59 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        return False
+
+    def get_by_category(self, category):
+        item_in_category = []
+        for item in self.inventory:
+            if item.category == category:
+                item_in_category.append(item)
+        
+        return item_in_category
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+
+        self.add(their_item)
+        vendor.add(my_item)
+        self.remove(my_item)
+        vendor.remove(their_item)
+        
+        return True
+
+    def swap_first_item(self, vendor_friend):
+        if not self.inventory or not vendor_friend.inventory:
+            return False
+
+        self.swap_items(vendor_friend, self.inventory[0], vendor_friend.inventory[0])
```

Comment:
Nice reuse of `swap_items`.

Notice, though, that by using `swap_items`, this winds up being O(n) (mostly due to removing the items from the inventory). We could simply swap the items in place here to have O(1) runtime with something like:
```python
        self.inventory[0], vendor_friend.inventory[0] = vendor_friend.inventory[0], self.inventory[0]
```

Alternatively, we could think about how to modify the implementation of `swap_items` so that it would still work in general, but also give us O(1) in this specific case. Try that out if that sounds interesting!